kbf-parse.js

```diff
@@ -0,0 +1,644 @@
+  KC_UNDERSCORE: 'KC_UNDS',
+  KC_LEFT_CURLY_BRACE: 'KC_LCBR',
+  KC_RIGHT_CURLY_BRACE: 'KC_RCBR',
+  KC_COLON: 'KC_COLN',
+  KC_DOUBLE_QUOTE: 'KC_DQUO',
+  KC_DQT: 'KC_DQUO',
+  KC_LEFT_ANGLE_BRACKET: 'KC_LT',
+  KC_LABK: 'KC_LT',
+  KC_RIGHT_ANGLE_BRACKET: 'KC_GT',
+  KC_RABK: 'KC_GT',
+  KC_QUESTION: 'KC_QUES',
+
+  // Unicode
+  UNICODE_MODE_FORWARD: 'UC_NEXT',
+  UNICODE_MODE_REVERSE: 'UC_PREV',
+  UC_MOD: 'UC_NEXT',
+  UC_RMOD: 'UC_PREV',
+  UC_M_OS: 'UC_MAC',
+  UC_M_MA: 'UC_MAC',
+  UC_M_LN: 'UC_LINX',
+  UC_M_WI: 'UC_WIN',
+  UC_M_BS: 'UC_BSD',
+  UC_M_WC: 'UC_WINC',
+  UNICODE_MODE_OSX: 'UC_MAC',
+  UNICODE_MODE_MAC: 'UC_MAC',
+  UNICODE_MODE_LNX: 'UC_LINX',
+  UNICODE_MODE_WIN: 'UC_WIN',
+  UNICODE_MODE_BSD: 'UC_BSD',
+  UNICODE_MODE_WINC: 'UC_WINC'
+}
+
+const mcus = ['atmega32u2', 'atmega32u4', 'at90usb1286']
+
+function parseProcessor(o) {
+  return mcus[o.controller]
+}
+
+function parseBootloader(o) {
+  switch (o.settings.bootloaderSize) {
+    case 0: // 512kB
+      return 'halfkay'
+    case 1: // 2048kB (should actually be 1024kB)
+      if (mcus[o.controller] == 'at90usb1286') {
+        return 'halfkay'
+      }
+      break
+  }
+
+  return 'atmel-dfu'
+}
+
+function parseDiodeDirection(o) {
+  switch (o.settings.diodeDirection) {
+    case 0:
+      return 'COL2ROW'
+    case 1:
+      return 'ROW2COL'
+  }
+}
+
+function parseBootmagic(o) {
+  if (o.settings.chooseBootmagic != null) {
+    // QMKeyboard.cn
+    return o.settings.chooseBootmagic != 1
+  } else if (o.settings.bootMagic != null) {
+    // mtkeyboard.vip
+    return !!o.settings.bootMagic
+  }
+
+  return true
+}
+
+function parseMousekey(o) {
+  if (o.settings.chooseMousekey != null) {
+    // QMKeyboard.cn
+    return !o.settings.chooseMousekey
+  } else if (o.settings.mouseKey != null) {
+    // mtkeyboard.vip
+    return !!o.settings.mouseKey
+  }
+
+  return true
+}
+
+function parseExtrakey(o) {
+  if (o.settings.chooseExtrakey != null) {
+    // QMKeyboard.cn
+    return !o.settings.chooseExtrakey
+  }
+
+  return true
+}
+
+function parseKeyLock(o) {
+  if (o.settings.choosekeylock != null) {
+    // QMKeyboard.cn
+    return !o.settings.choosekeylock
+  } else if (o.settings.keyLock != null) {
+    // mtkeyboard.vip
+    return !!o.settings.keyLock
+  }
+
+  return false
+}
+
+function parseFeatures(o) {
+  return {
+    bootmagic: parseBootmagic(o),
+    mousekey: parseMousekey(o),
+    extrakey: parseExtrakey(o),
+    console: false,
+    command: false,
+    nkro: false,
+    backlight: o.pins.led != null,
+    rgblight: o.pins.rgb != null,
+    audio: false,
+    key_lock: parseKeyLock(o)
+  }
+}
+
+function parseBacklight(o) {
+  return {
+    pin: o.pins.led,
+    levels: o.settings.backlightLevels,
+    breathing: false
+  }
+}
+
+function parseRGBlight(o) {
+  return {
+    pin: o.pins.rgb,
+    led_count: o.settings.rgbNum,
+    hue_steps: 8,
+    saturation_steps: 8,
+    brightness_step: 8,
+    max_brightness: '255',
```

Comment:
- `brightness_step` -> `brightness_steps`
- `max_brightness` needs to be an integer

```suggestion
    brightness_steps: 8,
    max_brightness: 255,
```